app/models/concerns/foreman_remote_execution/host_extensions.rb

```diff
@@ -26,8 +26,9 @@ def execution_status_label(options = {})
 
     def params_with_remote_execution
       params = params_without_remote_execution
-      keys = remote_execution_ssh_keys
-      params['remote_execution_ssh_keys'] = keys unless keys.blank?
+      if (keys = remote_execution_ssh_keys).present?
+        params['remote_execution_ssh_keys'] = params['remote_execution_ssh_keys'].to_s + keys.join('\n')
```

Comment:
`([params['remote_execution_ssh_keys'].to_s] + keys).join('\n')`, but let's talk to @stbenjam about this even being a good idea or not…